src/cryptonote_core/blockchain.cpp

```diff
@@ -3261,6 +3261,25 @@ bool Blockchain::check_tx_outputs(const transaction& tx, tx_verification_context
     return false;
   }
 
+  // from carrot or v18, require output pubkeys be sorted in strictly increasing lexicographical order
+  const bool tx_is_carrot = tx.vout.at(0).target.type() == typeid(txout_to_carrot_v1);
```

Comment:
Small nit since the rest of this function doesn't make any assumptions about `tx.vout` existing AFAICT:

```suggestion
  const bool tx_is_carrot = tx.vout.size() && tx.vout.at(0).target.type() == typeid(txout_to_carrot_v1);
```

Reply:
`check_output_types()` now checks that `vout` isn't empty, so this shouldn't ever trip, but it's probably good to do this to be defensive